icu4c/source/common/unicode/utypes.h

```diff
@@ -385,17 +385,30 @@ typedef double UDate;
 /*===========================================================================*/
 
 /**
- * Error code to replace exception handling, so that the code is compatible with all C++ compilers,
- * and to use the same mechanism for C and C++.
+ * Standard ICU4C error code class, a substitute for exceptions.
+ *
+ * Initialize the UErrorCode with U_ZERO_ERROR, and check for success or
+ * failure using U_FAILURE() or U_SUCCESS():
```

Comment:
confusing to have different orders of "success or failure" vs. "U_FAILURE() or U_SUCCESS()". Please swap the latter.